### PR TITLE
CI: Fix Coverage Analysis.

### PR DESCRIPTION
**Pull request by gbarkadiusz**
FIX: The coverage_analysis.py while generating report, duplicates some files and functions of components.
e.g for bluetooth.
image

After the fix.
image

### CODE REVIEW

File: scripts/ci/coverage/coverage_analysis.py [hunk 1]
```diff
@@ -284,6 +285,12 @@ def generate_json_report(self, coverage):
 
         return output_json
 
+    def is_duplicate(self, new_file, json_list):
+        for item in json_list:
+            if item['Name'] == new_file['Name']:
+                return True
+        return False
```
Comment (arikgreen) [inline scripts/ci/coverage/coverage_analysis.py:292]:
```suggestion
        res = [x for x in json_list if x['Name'] == new_file['Name']]
        return res
```

**Review by arikgreen — CHANGES_REQUESTED** [review]

File: scripts/ci/coverage/coverage_analysis.py [hunk 2]
```diff
@@ -274,7 +274,8 @@ def generate_json_report(self, coverage):
                                                 "Name":i_fun['name']
                                             }
                                             json_file['Uncovered_Functions'].append(json_uncovered_funciton)
-                                    json_files.append(json_file)
+                                    if not self.is_duplicate(json_file, json_files):
```
Comment (arikgreen) [inline scripts/ci/coverage/coverage_analysis.py:277]:
```suggestion
                                    comp_exists = [x for x in json_files if x['Name'] == json_file['Name']]
                                    if not comp_exists:
```

File: scripts/ci/coverage/coverage_analysis.py [hunk 3]
```diff
@@ -284,6 +285,12 @@ def generate_json_report(self, coverage):
 
         return output_json
 
+    def is_duplicate(self, new_file, json_list):
+        for item in json_list:
+            if item['Name'] == new_file['Name']:
+                return True
+        return False
+
```
Comment (arikgreen) [inline scripts/ci/coverage/coverage_analysis.py:293]:
```suggestion
   
```

**Review by arikgreen — CHANGES_REQUESTED** [review]

**Comment by gbarkadiusz** [comment]
@arikgreen  Please take a look again.

**Review by arikgreen — DISMISSED** [review]

**Review by arikgreen — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**